### ISSUE TITLE
feature request: secrets publishing

### ISSUE THREAD

**Issue by Cherry** [cloudflare/wrangler-action#19]
We maintain secrets in GitHub Actions for different environments, and now with the [new secrets release](https://blog.cloudflare.com/workers-secrets-environment/) available in Workers, it would be fantastic to be able to publish secrets directly using this action.

Something like:

```yaml
jobs:
  deploy:
    runs-on: ubuntu-latest
    name: Deploy
    steps:
      - uses: actions/checkout@master
      - name: Publish
        uses: cloudflare/wrangler-action@1.1.0
        with:
          apiToken: ${{ secrets.CF_API_TOKEN }}
          environment: 'production'
          secrets:
            - foo: ${{ secrets.production_foo }}
            - bar: ${{ secrets.production_bar }}
```

This would allow us to set and deploy secrets in GitHub actions for different environments, without having someone manually issue `wrangler secret` commands, or manually creating our own step where we `npm install` wrangler, and run the commands inline.

**Comment by tillkruss** [comment]
This would be huge 👍🏻

**Comment by tillkruss** [comment]
Might be relevant to #19.

**Comment by signalnerve** [comment]
hey! love this. the idea is basically passing input, whether that's a secret or plain text, from the action into wrangler and then into your app. 

the one problem here is that there isn't necessarily a 1 to 1 mapping between secrets and what we traditionally think of as `process.env` or env vars – something like `STRIPE_SECRET` would come from an explicit `wrangler secret put STRIPE_SECRET` or by setting the [vars](https://developers.cloudflare.com/workers/tooling/wrangler/configuration#per-project) field in your `wrangler.toml`. 

i think my gut feeling here is that having the `secrets` field makes sense – the problem is that `wrangler secret put` is an interactive command, so there might be some shell trickiness to input that correctly as the command is currently implemented. 

an alternative would be wrangler supporting `.env`-style files more generally – then maybe we can build out a separate workflow (or find an existing one) for transforming env vars/secrets in github actions to a `.env` file. could be more broadly applicable to applications on github actions as a platform (and not over-crowding this action)??

cc-ing the developer productivity team here (@cloudflare/workers-devexp) as a potential solution here is almost in "needs wrangler support" territory and i don't want to patch around it instead of potentially building a better solution

**Comment by ashleymichal** [comment]
> wrangler secret put is an interactive command

pipes  are  great for this ;)

`cat secret.txt | wrangler secret put` or `echo $SECRET_ENV_VAR | wrangler  secret put` may get you where you want to go?

**Comment by tillkruss** [comment]
Setting the `vars` based on `with` properties would be quite easy, eh?

**Comment by ashleymichal** [comment]
> And then append the vars line to the wrangler.toml.

a simple append will not be appropriate across all invocations, since one wrangler.toml can specify multiple environments and the vars would be appended only to the last specified env.

**Comment by signalnerve** [comment]
closed with #28. woo!
